specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1178 @@
+  </visual>
+
+  <actions>
+    <input id="ComboBox" type="selection" defaultInput="yes">
+      <selection id="yes" content="Going" />
+      <selection id="maybe" content="Maybe" />
+      <selection id="no" content="Decline" />
+    </input>
+    <action
+      arguments="action=Send"
+      hint-inputId="ComboBox"
+      content="Send"/>
+
+  </actions>
+</toast>
+```
+
+# AppNotificationProgressBar
+
+AppNotificationProgressBar is a component that builds the `<progress>` xml.
+AppNotificationProgressBar is used to update an AppNotification that is currently displayed. This
+allows developers to inform users about the status of certain operations such as download or
+install.
+
+**WinAppSDK 1.2 `<progress>` Schema**:
+
+```xml
+<progress title? = string
+    status = string
+    value = string
+    valueStringOverride? = string
+/>
+```
+
+AppNotificationProgressBar uses data binding to update the active UI. In the WindowsAppSDK, this is
+done by using AppNotificationProgressData. Click
+[Here](https://github.com/microsoft/WindowsAppSDK/blob/main/specs/AppNotifications/AppNotifications-spec.md#notification-progress-updates)
+for an example.
+
+All the attributes have static data binding constants:
+
+-   Status = "{progressStatus}"
+-   Value = "{progressValue}"
+-   Title = "{progressTitle}"
+-   ValueStringOverride = "{progressValueString}
+
+Every AppNotificationProgressBar component will have the bound status and value. Title and
+ValueStringOverride are not required so those constants will not be added unless you call
+the AppNotificationProgressBar component APIs
+
+-   UseTitleForLabel(String title)
+-   BindTitleForLabel()
+-   BindValueStringOverride()
+
+Below is an example use:
+
+```cpp
+// Using AppNotificationContent, build the Xml payload with AppNotificationProgressBar.
+winrt::hstring xmlPayload { AppNotificationContent()
+    .AddText(L"Downloading this week's new music...")
+    .AddProgressBar(AppNotificationProgressBar()
+        .BindTitleForLabel()
+        .BindValueStringOverride())
+    .GetXml() };
+```
+
+Xml output:
+
+```xml
+<toast>
+  <visual>
+    <binding template="ToastGeneric">
+      <text>Downloading this week's new music...</text>
+      <progress
+        title="{progressTitle}"
+        value="{progressValue}"
+        valueStringOverride="{progressValueString}"
+        status="{progressStatus}"/>
+    </binding>
+  </visual>
+</toast>
+```
+
+Construct an AppNotification with the string xml payload from AppNotificationContent
+
+```cpp
+winrt::Microsoft::Windows::AppNotifications::AppNotification notification(xmlPayload);
+notification.Tag(L"Tag");
+notification.Group(L"Group");
+```
+
+Assign initial values for first notification progress UI:
+
+```cpp
+winrt::Microsoft::Windows::AppNotifications::AppNotificationProgressData data(1 /* Sequence number */);
+data.Title(L"Katy Perry"); // Binds to {progressTitle} in xml payload
+data.Value(0.5); // Binds to {progressValue} in xml payload
```

Comment:
How do I set the value to `indeterminate`? Do I use NaN?

```cpp
data.Value(std::numeric_limits<double>::quiet_NaN());
```

Reply:
Consider for the future in WinAppSDK.

Reply:
There is already an issue tracking this: https://github.com/microsoft/WindowsAppSDK/issues/2231